templates: Revise linkifier and code playground explanatory text in org settings overlay.

laurynmm opened the pull request:
Revises descriptive text and examples at the top of the linkifiers and code playgrounds tabs in the organization settings overlay. See [relevant CZO discussion](https://chat.zulip.org/#narrow/stream/101-design/topic/descriptions.20on.20linkifiers.20and.20code.20playground.20tabs/near/1511344). 

**Notes**:
- Adds the `rendered_markdown` class to the HTML code elements, via an HTML span element so that the specific CSS rules for code elements with that class will be applied to these examples. See [relevant comment in pull request](https://github.com/zulip/zulip/pull/24430#issuecomment-1438925792).

---

**Screenshots and screen captures:**

<details>
<summary>Linkifiers tab - narrowed for before and after comparison</summary>

| Before | After |
| --- | --- |
| ![image](https://user-images.githubusercontent.com/63245456/221880402-fc7a170f-b695-40f5-a04b-0ae7fce88cde.png) | ![image](https://user-images.githubusercontent.com/63245456/221880471-328f65b8-367b-47c3-9d2d-45331f543a16.png)
</details>
<details>
<summary>Code playgrounds tab - narrowed for before and after comparison</summary>

| Before | After |
| --- | --- |
| ![image](https://user-images.githubusercontent.com/63245456/221880786-ad736398-6f86-461f-89c3-74f6a0bfec5e.png) | ![image](https://user-images.githubusercontent.com/63245456/221880870-da7685f2-1faf-4285-8059-d531f9d49d69.png)
</details>

<details>
<summary>Linkifiers tab - updated version in both themes, not narrowed</summary>

![Screenshot from 2023-02-28 15-22-16](https://user-images.githubusercontent.com/63245456/221881716-72ea3abe-40c6-4b1e-9d0c-abdd4a992a74.png)
![Screenshot from 2023-02-28 15-22-28](https://user-images.githubusercontent.com/63245456/221881720-8d82db50-ea03-4349-906d-c7f09554d2c0.png)
</details>

<details>
<summary>Code playgrounds tab - updated version in both themes, not narrowed</summary>

![Screenshot from 2023-02-28 15-22-52](https://user-images.githubusercontent.com/63245456/221881853-2b2df405-a4ae-4123-8b2f-9878550b37a0.png)
![Screenshot from 2023-02-28 15-23-02](https://user-images.githubusercontent.com/63245456/221881859-3fb33d10-494c-4538-868b-587b212b6980.png)

</details>

---

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

alya commented on web/templates/settings/linkifier_settings_admin.hbs:
```diff
@@ -3,40 +3,29 @@
-            project, you could use the following:
+            Configure regular expression patterns that will be used to
+            automatically transform any matching text in message topics
+            and content into links.
```
How about:

> .. any matching text in Zulip messages and topics into links.

laurynmm replied:
Done.

alya commented on web/templates/settings/linkifier_settings_admin.hbs:
```diff
@@ -3,40 +3,29 @@
+            For example, to automatically linkify issue numbers (e.g. #123)
+            to the corresponding items in a GitHub project, you could use the
+            following regular expression pattern and URL format string:
             {{/tr}}
```
Maybe we should just reuse the text on the help center page for the second paragraph here?

> Linkifiers make it easy to refer to issues or tickets in third party issue trackers, like GitHub, Salesforce, Zendesk, and others. For instance, you can add a linkifier that automatically turns #2468 into a link to https://github.com/zulip/zulip/issues/2468.

laurynmm replied:
Note that GitHub is linking to the issue here, but in the overlay it would just be text.

laurynmm replied:
See screenshot in [update post/comment](https://github.com/zulip/zulip/pull/24513#issuecomment-1450436639) below.

alya replied:
Oh, I wasn't proposing cutting the bullets. Dropping the URL from the paragraph seems fine; I would cut "related" from your proposed text.

alya commented:
Side comment: The placement of the "Add X" button looks odd in the narrow screenshots. Should we file a follow-up issue to make it centered when it's on its own line?

alya commented on web/templates/settings/playground_settings_admin.hbs:
```diff
@@ -2,24 +2,35 @@
+                such as <z-link-repl>replit</z-link-repl>, so that <z-link-code-blocks>code blocks</z-link-code-blocks>
+                that are tagged with a programming language will have a button visible on
+                hovering that allows users to open the code block on the code playground site.
+                {{#*inline "z-link-code-playgrounds"}}<a href="/help/code-blocks#code-playgrounds" target="_blank" rel="noopener noreferrer">{{> @partial-block}}</a>{{/inline}}
```
I'm not sure these changes make the intro more clear, and the current intro matches the /help page, which is nice. We could probably drop the "Configure external code playgrounds for your Zulip organization." intro sentence; I don't think it's doing much work here.

alya replied:
I'd be open to any changes that can be made identically here on the help center page and help make things more clear and/or succinct.

laurynmm replied:
I'll cut that first sentence ("Configure external ..."), so that the text in the overlay is the two main sentences in the help center article.

I understand wanting to keep these similar, but I'll note that if we update/edit the help center text in the future, I can see us missing that this text would also need to be updated/reviewed for those changes.

laurynmm replied:
See screenshot in [update post/comment](https://github.com/zulip/zulip/pull/24513#issuecomment-1450436639) below.

alya commented on web/templates/settings/playground_settings_admin.hbs:
```diff
@@ -2,24 +2,35 @@
-            and <i>URL prefix</i> fields as:
+            For example, to configure a code playground for code blocks tagged with
+            JavaScript, you could specify the language and URL prefix fields as:
             {{/tr}}
```
How about just this?

> For example, to configure a code playground for code blocks tagged as JavaScript, set:

laurynmm replied:
Done, except now the example is Python.

alya commented on web/templates/settings/playground_settings_admin.hbs:
```diff
@@ -2,24 +2,35 @@
+            The name field appears in the tooltip shown when a user hovers on
+            the button in the code block.
+            {{/tr}}
+        </p>
```
I think we should use the same example as in the placeholder text. Otherwise, you can have the annoying experience that you want the playground in the placeholder, but don't have a copy-paste way to set it up.

laurynmm replied:
Done.

alya commented on web/templates/settings/playground_settings_admin.hbs:
```diff
@@ -2,24 +2,35 @@
+        <p>
+            {{#tr}}
+            The name field appears in the tooltip shown when a user hovers on
+            the button in the code block.
```
I think we should include "Language:" above for clarity/completeness, and then this  note can get moved to parenthetical about the language field.

laurynmm replied:
I think you meant "Name" here. I added it to the bullet point example.

laurynmm replied:
See screenshot in [update post/comment](https://github.com/zulip/zulip/pull/24513#issuecomment-1450436639) below.

laurynmm commented:
> Side comment: The placement of the "Add X" button looks odd in the narrow screenshots. Should we file a follow-up issue to make it centered when it's on its own line?

Yeah, that placement is a bit odd. There are a lot of specific CSS rules for these forms/inputs/buttons. When looking at the CSS rules, I noticed in [this commit](https://github.com/zulip/zulip/commit/e30e8c0e6d304c39cef67003dd723a30518378a1) there was a mention of maybe moving them to a modal? But I didn't see any issues for that. These are the only two of the settings overlay tabs that have the form/input in the overlay body.

---

**Updates**:
- Text revised for majority of review comments above. See screenshots below for current PR version.

**Updated screenshots**:

<details>
<summary>Linkifiers</summary>

![Screenshot from 2023-03-01 17-09-24](https://user-images.githubusercontent.com/63245456/222199670-1151e401-22d4-4658-b3a1-73b848c3854e.png)
</details>
<details>
<summary>Code playgrounds</summary>

![Screenshot from 2023-03-01 17-09-28](https://user-images.githubusercontent.com/63245456/222199606-347fe554-06cb-44d5-a4cb-2dda3afc6cba.png)
</details>

alya commented:
Looks good to me other than the one-word change noted above!

I think we should also point the user to the help center article for further help with code playgrounds, the way we do for linkifiers. Maybe something like:

> For more examples and technical details, see the [help center documentation](/help/code-blocks#code-playgrounds) on adding code playgrounds.

laurynmm commented:
**Updates**:
- Made the one-word change noted above (remove "related").
- Added the extra sentence to code playgrounds tab with the link, and made that text/link match better between the two tabs.
- Added an icon help center circle link to the linkifiers form/input title, which matches the code playgrounds and means it will be there if these are pulled out to be modals. Alternatively, we could remove the icon help center link in code playgrounds.

**Updated screenshots**:

<details>
<summary>Linkifiers</summary>

![Screenshot from 2023-03-02 18-05-17](https://user-images.githubusercontent.com/63245456/222501134-71c69b54-c99f-49bb-bdd7-3ae7617e0864.png)
</details>
<details>
<summary>Code playgrounds</summary>

![Screenshot from 2023-03-02 18-05-25](https://user-images.githubusercontent.com/63245456/222501330-f19db2db-6ea3-4e23-8b08-51c823070b69.png)
</details>

alya commented:
Works for me! @timabbott can you take a look?

timabbott commented:
Looks great, merged, thanks @laurynmm!
